src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.command.Command;
+import duke.exception.InvalidInstructionFormatException;
+import duke.exception.InvalidInstructionLengthException;
+import duke.exception.InvalidTaskIndexException;
+import duke.exception.MissingFieldException;
+import duke.exception.TaskDoneException;
+import duke.exception.UnknownInstructionException;
+import duke.logic.StorageManager;
+import duke.logic.TaskList;
+import duke.logic.UiManager;
+import duke.logic.UserInputParser;
+
+/**
+ * Represents a Duke Chat bot.
+ * It contains a <code>TaskList</code> to track the user's tasks,
+ * a <code>UIManager</code> to handle user interactions and
+ * a <code>StorageManager</code> to handle storing of data.
+ */
+public class Duke {
+    private final UiManager uiManager;
+    private final StorageManager storageManager;
+    private TaskList taskList;
+
+    public Duke() {
+        this.uiManager = new UiManager();
+        this.storageManager = new StorageManager(CommonString.DUKE_FILE_PATH.toString());
+        try {
+            this.taskList = new TaskList(storageManager.loadData());
+        } catch (FileNotFoundException e) {
+            System.out.println("Cannot Load Data: " + e.getMessage());
+            e.printStackTrace();
+            taskList = new TaskList(new ArrayList<>());
+        }
+    }
+
+
+    /**
+     * Executes Main method for Duke
+     * Initialises <code>Duke</code> object and runs.
+     */
+    public static void main(String[] args) {
+        // Initialisation of duke.Duke
+        Duke duke = new Duke();
+        duke.run();
+    }
+
+    /**
+     * Executes Duke Program.
+     * User input is extracted by <code>UIManager</code> and parsed by <code>UserInputParser</code>,
+     * generating a command. Execution of commands are managed by individual <code>Command</code> instances.
+     * Storing/Loading of data is managed by <code>StorageManager</code>.
+     */
+    public void run() {
+        // INTRO
+        uiManager.printDukeIntro();
+        boolean isExit = false;
+
+        // Execute duke.Duke Functions
+        while (!isExit) {
+            try {
+                String userInput = uiManager.readCommand();
+                Command command = UserInputParser.parse(userInput);
+                command.execute(taskList, uiManager, storageManager);
```

Comment:
As with the taskList being passed around the entire program, the uiManager and storageManager Object are being passed around the entire program, which may not be necessary. Consider using a Non-Persistent Storage Object at initialisation of program so that you can easily reference to the non persistent data throughout the program without using up parameter space to pass in objects that may or may not be used by the methods/classes that they are being passed into